root/etc/e-smith/events/actions/nethserver-directory-user-unlock

```diff
@@ -36,6 +36,14 @@ if( ! $userName ) {
 
 $userName = (split(/@/,$userName))[0];
 
+# Do not unlock users without password
+my $currentPassword = `/usr/bin/ldapsearch -o ldif-wrap=no -LLL -Y EXTERNAL '(uid=$userName)' userPassword 2>/dev/null | /usr/bin/grep userPassword | /usr/bin/cut -d':' -f 3 | /usr/bin/base64 -id`;
+chomp $currentPassword;
+if ($currentPassword eq '{CRYPT}!!') {
```

Comment:
- Simplify the long Bash commands pipeline (base64 and cut are not necessary)
- Let's expect the two possible encodings LDIF can provide

```suggestion
my $currentPassword = qx(/usr/bin/ldapsearch -o ldif-wrap=no -LLL -Y EXTERNAL '(uid=$userName)' userPassword 2>/dev/null | /usr/bin/grep ^userPassword);
chomp $currentPassword;
if ($currentPassword eq 'userPassword: {CRYPT}!!' || $currentPassword eq 'userPassword:: e0NSWVBUfSEh') {
```

Reply:
I agree, but why the check on  2 different strings? I think that `$currentPassword eq 'userPassword:: e0NSWVBUfSEh'` should be enough.

Reply:
Yes it should...

Writing it explicitly can also be seen as "self-docs"

Reply:
I'd like to make it explicit.
What about this?
```suggestion
my $currentPassword = qx(/usr/bin/ldapsearch -o ldif-wrap=no -LLL -Y EXTERNAL '(uid=$userName)' userPassword 2>/dev/null | /usr/bin/grep ^userPassword);
chomp $currentPassword;

# e0NSWVBUfSEh is base64 form of empty password: {CRYPT}!!
if ($currentPassword eq 'userPassword:: e0NSWVBUfSEh') {
```